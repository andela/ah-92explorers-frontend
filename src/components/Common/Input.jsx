/* eslint-disable react/require-default-props */

import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const Input = ({
  name, value, error, type, placeholder, onChange,
}) => (
  <div className={classnames('form-group', { 'is-invalid': error })}>
    <input
      onChange={onChange}
      value={value}
      type={type}
      name={name}
      className="form-control"
      placeholder={placeholder}
    />
    {error && <span className="text-danger">{error}</span>}
  </div>
);

Input.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  error: PropTypes.string,
  type: PropTypes.string.isRequired,
};

Input.defaultProps = {
  // eslint-disable-next-line react/default-props-match-prop-types
  type: 'text',
};

export default Input;
