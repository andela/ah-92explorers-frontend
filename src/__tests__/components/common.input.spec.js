import '@babel/polyfill';
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'
import configureMockStore from 'redux-mock-store';
import promiseMiddleware from 'redux-promise-middleware';
import thunk from 'redux-thunk';
import Input from '../../components/Common/Input';

const middlewares = [thunk, promiseMiddleware];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});
const props = {
  getInfo: jest.fn(),
  onChange: jest.fn(),
  name: 'input',
  value: 'hola',
  type: 'text',
};
describe('Render Common Input component', () => {
  it('to have wrapper class', async () => {
    const wrapper = mount(<MemoryRouter><Provider store={store}><Input {...props} /></Provider></MemoryRouter>);
    expect(wrapper).toMatchSnapshot();
  });
});
