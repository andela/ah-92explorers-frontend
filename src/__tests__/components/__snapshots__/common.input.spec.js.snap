// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Common Input component to have wrapper class 1`] = `ReactWrapper {}`;
